shared/constants/security-provider.ts

```diff
@@ -1,3 +1,21 @@
+export enum SecurityProvider {
+  Blockaid = 'blockaid',
+}
+
+/**
+ * @typedef {object} SecurityProviderConfig
+ * @property {string} tKeyName - translation key for security provider name
+ * @property {string} url - URL to securty provider website
+ */
+
+/** @type {Record<string, SecurityProviderConfig>} */
+export const SECURITY_PROVIDER_CONFIG = {
+  [SecurityProvider.Blockaid]: {
+    tKeyName: 'blockaid',
+    url: 'https://blockaid.io/',
+  },
+};
```

Comment:
Besides the jsdoc why not create actual type and benefit from the full extent of TS? You could have something like:
```js
type SecurityProviderConfig = Record<
  SecurityProvider,
  {
    tKeyName: string;
    url: string;
  }
>;
```

Reply:
lgtm! I was following patterns from `shared/constants/gas.ts` and other files. updated here https://github.com/MetaMask/metamask-extension/pull/19825/commits/fac74730f8d9d889be872cde88bb9cfcfdcbb7b2

does this look okay or are there any more suggestions? 

cc: @cryptotavares

Reply:
[nit]
Do we want this const to be readonly (at least for TS)? If yes then we can:
```js
const SECURITY_PROVIDER_CONFIG: Readonly<SecurityProviderConfig>
```
or ditch the type and use const assertion ([TS docs for readonly and const assertion](https://www.typescriptlang.org/docs/handbook/typescript-in-5-minutes-func.html#readonly-and-const))
```js
export const SECURITY_PROVIDER_CONFIG = {
  [SecurityProvider.Blockaid]: {
    tKeyName: 'blockaid',
    url: 'https://blockaid.io/',
  },
} as const;
```

Using the const assertion seems more inline with what is being done on other constants TS files ([example](https://github.com/MetaMask/metamask-extension/blob/develop/shared/constants/app.ts#L110)).

(Sorry for asking to add the type and then replacing it for const assertion.. I am mostly used to use the Readonly mapped type, but that does not seem to be the standard on the Extension code 😅 )

Reply:
hey @cryptotavares, thanks for the suggestion and references! I applied the const assertion and I am thinking the type you suggested earlier is also nice to help define the types for each of the config objects. (Before, I was using jsdoc @typedef instead of the type you first proposed.)

Reply:
If you use the const assertion TS should automatically pick the correct type.
If you do const assertion and type the const at the same time, then TS will interpret the explicit type as the actual type (meaning that if you do not add the readonly utility type, then it will allow overriding the current property values). Example:
```js
// Just using const assertion
const test = {
  yolo: 1
} as const;

test.yolo = 2; // TS will complain about this because when we asserted to const, test is readonly
```

```js
type Test = {
  yolo: number
}

// explicit typing with const assertion
const test: Test = {
  yolo: 1
} as const;

test.yolo = 2; // TS will accept this, as it considers that test.yolo is a number and is not readonly
```
Anyway, I was suggesting that you can either use const assertion or you can explicitly type it (and use the Readonly utility type). But as const assertion seems to be the pattern used in the rest of the codebase you can do just that.

Btw, other examples on how we can force it to be readonly with explicit typing:
```js
// Using Readonly utility type ensures that all properties are readonly and cannot be reassigned. 
const test: Readonly<Test> = {
  yolo: 1
};
```
```js
const test: Test = {
  // using readonly modifier ensures that that specific property is readonly and cannot be reassigned
  readonly yolo: 1
};
```

Reply:
ooh I see now. I think I got it https://github.com/MetaMask/metamask-extension/pull/19825 cc: @cryptotavares